### DISCUSSION TITLE
Versions UI

### DISCUSSION THREAD

**Discussion by bosschaert** [adobe/da-live#87] [Ideas]
I'm thinking about a versioning UI where you can see the versions of the current document, restore to a given version and have a way to explicitly create a version from the current state (it should also be possible to get versions created automatically in the background).

One way could be to add a _versions_ button alongside the current _preview_ button:
<img width="1678" alt="Screenshot 2024-03-20 at 16 33 20" src="https://github.com/adobe/da-live/assets/226514/82c99402-8ca1-46e7-bd9e-a7ebf548fd3b">

When you click it you get a versions panel to the right, similar to what the preview panel looks like. It has it's own button bar on the right hand side where you can create a new version from the current state of the document.
You can also click the versions in the list to see the content of them (overlays the main editor with the content of the selected version) and click the 'restore' button to restore the selected version as the current version. 

<img width="1678" alt="Screenshot 2024-03-20 at 16 33 31" src="https://github.com/adobe/da-live/assets/226514/7d45d777-038d-47a6-998e-cc6ff4239996">

Thoughts anyone?

(PS the icons in these screenshots are just temporary)

**Comment by auniverseaway** [comment]
I think this is great. I like the idea that the right pane is the section of the page where you can see different views into your doc: live preview, versions, other comparisons, etc. The only thing I would do is de-emphasize the button under the preview. 

If I think about how much each are going to be used, live preview will be used an order of magnitude more than the versions button. I have also seen in on-boarding, it takes people 1-2s to see where live preview is. If they have to visually sort through icons (I can see us adding more here), it will take longer.

At least to start, I think the best of both worlds would be to simply have a "quiet" icon below the live preview tab. This will help so it competes less for the visual attention of live preview. Quiet would mean just a floating icon by itself below the live preview tab.

Would also love to hear @docs-dre's or Rikio's thoughts on this.

**Comment by auniverseaway** [comment]
I'm not real worried about it today, but there is a possibility that we paint ourselves into a corner if we want to quickly click through versions while also seeing the live preview. Until we have a better idea and see how people are using this, I think the above is a good starting place.
